mediacapture-streams/MediaDevices-IDL-all.html

```diff
@@ -0,0 +1,49 @@
+<!doctype html>
+<html>
+  <head>
+    <title>getUserMedia: Non-Interactive test for mediaDevices APIs</title>
+    <link rel="author" title="Dr Alex Gouaillard" href="mailto:agouaillard@gmail.com"/>
+    <link rel="help" href="http://w3c.github.io/mediacapture-main/getusermedia.html#mediadevices">
+    <link rel="help" href="http://w3c.github.io/mediacapture-main/getusermedia.html#mediadevices-interface-extensions">
+    <meta name='assert' content='Check the mediaDevices APIs.'/>
+  </head>
+  <body>
+    <h1 class="instructions">Description</h1>
+    <p class="instructions">This test checks for the presence of the
+    <code>navigator.mediaDevices.getUserMedia</code> method.</p>
+    <div id='log'></div>
+    <script src=/resources/testharness.js></script>
+    <script src=/resources/testharnessreport.js></script>
+    <script src=/resources/WebIDLParser.js></script>
+    <script src=/resources/idlharness.js></script>
+    <script>
+      var idl_array = new IdlArray();
+
+      idl_array.add_untested_idls("interface EventTarget {\
+        void addEventListener(DOMString type, EventListener? callback, optional boolean capture = false);\
+        void removeEventListener(DOMString type, EventListener? callback, optional boolean capture = false);\
+        boolean dispatchEvent(Event event);\
+      };");
+
+      idl_array.add_untested_idls("interface Navigator{\
+        readonly attribute MediaDevices mediaDevices;\
+      };");
+
+      idl_array.add_idls("[Exposed=Window]\
+        interface MediaDevices : EventTarget {\
+          attribute EventHandler ondevicechange;\
+          Promise<sequence<MediaDeviceInfo>> enumerateDevices ();\
+      };");
+
+      // TODO: fix the constraints argument (DrAlex)
```

Comment:
Which failure are you getting? In firefox, I get a failure about `navigator.mediaDevices.getUserMedia.length` being wrong (0 when it should be 1), which seems like a bug in firefox (a surprising one though, given their use of WebIDL processor). My version of Chrome doesn't have `navigator.mediaDevices.getUserMedia` at all, so it can only fail.